frontend/src/components/search/SearchBar.js

```diff
@@ -0,0 +1,138 @@
+import React, { Component } from "react";
+import { bindActionCreators } from "redux";
+import { connect } from "react-redux";
+import { Input, Dropdown, Card, Col } from "antd";
+import { getSearchBarResults } from "@/selectors/searchSelectors";
+import { fetchSearchBarResults, clearSearchBarResults } from "@/actionCreators/searchActionCreator";
+import PropTypes from "prop-types";
+import { withRouter } from "react-router-dom";
+import { SearchBarDropdown } from "@/components/search/SearchBarDropdown";
+import { throttle } from "lodash";
+
+const { Search } = Input;
+
+const propTypes = {
+  history: PropTypes.shape({ push: PropTypes.func }).isRequired,
+  fetchSearchBarResults: PropTypes.func.isRequired,
+  clearSearchBarResults: PropTypes.func.isRequired,
+  searchBarResults: PropTypes.arrayOf(PropTypes.object).isRequired,
+};
+
+const defaultPlaceholderText = "Search";
+const selectedPlaceholderText = "Type to search for mines, contacts, or more";
+
+export class SearchBar extends Component {
+  state = {
+    isSelected: false,
+    searchTerm: "",
+    searchTermHistory: [],
+  };
+
+  constructor(props) {
+    super(props);
+    this.fetchSearchResultsThrottled = throttle(this.props.fetchSearchBarResults, 2000, {
+      leading: true,
+      trailing: true,
+    });
+  }
+
+  changeSearchTerm = (e) => {
+    const newSearchTerm = e.target.value;
+    this.setState({ searchTerm: newSearchTerm });
+
+    if (newSearchTerm.length > 2) {
+      this.fetchSearchResultsThrottled(newSearchTerm);
+    }
+  };
+
+  search = (searchTerm) => {
+    if (searchTerm) {
+      this.setState((prevState) => {
+        const newSearchTermHistory = prevState.searchTermHistory.slice(
+          Math.max(prevState.searchTermHistory.length - 2, 0)
+        );
+        newSearchTermHistory.push(searchTerm);
+        return {
+          isSelected: false,
+          searchTermHistory: newSearchTermHistory,
+        };
+      });
+      this.clearSearchBar(false);
```

Comment:
`this.clearSearchBar()` can be updated to accept a param, if the param isn't passed in it is implied to be false. 

```
clearSearchBar = (isSelected) => {
    this.props.clearSearchBarResults();
    this.setState({
      isSelected,
      searchTerm: "",
    });
  };
```